packages/help-center/src/components/help-center.tsx

```diff
@@ -2,51 +2,136 @@
  * Internal Dependencies
  */
-import { useHCWindowCommunicator } from '../happychat-window-communicator';
+import useMessagingAuth from '../hooks/use-messaging-auth';
 import { useStillNeedHelpURL } from '../hooks/use-still-need-help-url';
+import useZendeskConfig from '../hooks/use-zendesk-config';
 import { HELP_CENTER_STORE, USER_STORE, SITE_STORE } from '../stores';
 import { Container } from '../types';
 import HelpCenterContainer from './help-center-container';
 import type { HelpCenterSelect, SiteSelect, UserSelect } from '@automattic/data-stores';
 import '../styles.scss';
 
+const ZENDESK_SCRIPT_ID = 'ze-snippet';
+
 const HelpCenter: React.FC< Container > = ( { handleClose, hidden } ) => {
 	const portalParent = useRef( document.createElement( 'div' ) ).current;
 	const { data: chatStatus } = useSupportAvailability( 'CHAT' );
-	const { data } = useHappychatAvailable( Boolean( chatStatus?.is_user_eligible ) );
-	const { setShowHelpCenter } = useDispatch( HELP_CENTER_STORE );
-	const { setUnreadCount } = useDispatch( HELP_CENTER_STORE );
 	const { setSite } = useDispatch( HELP_CENTER_STORE );
+	const { setShowMessagingLauncher } = useDispatch( HELP_CENTER_STORE );
+	const { setShowMessagingWidget } = useDispatch( HELP_CENTER_STORE );
+	const { setThirdPartyCookiesAllowed } = useDispatch( HELP_CENTER_STORE );
 
-	const { show, isMinimized } = useSelect(
-		( select ) => ( {
-			isMinimized: ( select( HELP_CENTER_STORE ) as HelpCenterSelect ).getIsMinimized(),
-			show: ( select( HELP_CENTER_STORE ) as HelpCenterSelect ).isHelpCenterShown(),
-		} ),
-		[]
-	);
+	const { status: zendeskStatus } = useZendeskConfig( Boolean( chatStatus?.is_user_eligible ) );
+	useEffect( () => {
+		if ( zendeskStatus === 'success' ) {
+			setThirdPartyCookiesAllowed( true );
+		} else if ( zendeskStatus === 'error' ) {
+			setThirdPartyCookiesAllowed( false );
+		}
+	}, [ setThirdPartyCookiesAllowed, zendeskStatus ] );
+
+	useEffect( () => {
+		if ( ! chatStatus?.is_user_eligible ) {
+			return;
+		}
+
+		const zendeskKey: string | false = config( 'zendesk_support_chat_key' );
+		if ( ! zendeskKey ) {
+			return;
+		}
+
+		if ( document.getElementById( ZENDESK_SCRIPT_ID ) ) {
+			return;
+		}
+
+		function setUpMessagingEventHandlers() {
+			if ( typeof window.zE !== 'function' ) {
+				return;
+			}
+
+			window.zE( 'messenger', 'hide' );
+
+			window.zE( 'messenger:on', 'open', function () {
+				setShowMessagingWidget( true );
+			} );
+			window.zE( 'messenger:on', 'close', function () {
+				setShowMessagingWidget( false );
+			} );
+			window.zE( 'messenger:on', 'unreadMessages', function ( count ) {
+				if ( Number( count ) > 0 ) {
+					setShowMessagingLauncher( true );
+				}
+			} );
+		}
 
-	const { unreadCount, closeChat } = useHCWindowCommunicator( isMinimized || ! show );
+		loadScript(
+			'https://static.zdassets.com/ekr/snippet.js?key=' + encodeURIComponent( zendeskKey ),
+			setUpMessagingEventHandlers,
+			{ id: ZENDESK_SCRIPT_ID }
+		);
```

Comment:
This would be nice to be wrapped in a factory function in the Help Center to avoid repetition. Something like

`ZDInstance({ onUnread, onClose, onOpen })`

Reply:
Yup, absolutely! I'll definitely clean up this code once this is launched and any major bugs are fixed. I want to keep it in an "exploded" format for now, so that it's easier to move things around and quickly refactor things 🙇‍♂️